tests/test_brightStarStamps.py

```diff
@@ -0,0 +1,143 @@
+# This file is part of meas_algorithms.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import unittest
+import numpy as np
+import os
+
+from lsst.meas.algorithms import brightStarStamps
+from lsst.afw import image as afwImage
+from lsst.daf.base import PropertySet
+import lsst.utils.tests
+
+
+class BrightStarStampsTestCase(lsst.utils.tests.TestCase):
+    """Test BrightStarStamps.
+    """
+    def setUp(self):
+        stampSize = (25, 25)
+        # create dummy star stamps
+        starImages = [afwImage.maskedImage.MaskedImageF(*stampSize)
+                      for _ in range(3)]
+        for starIm in starImages:
+            starImArray = starIm.getImage().getArray()
+            starImArray += np.random.rand(*stampSize)
+        ids = ["111", "aaa", "bbb"]
+        mags = np.random.rand(3)
+        # faint object to test magnitude cuts
+        self.faintObjIdx = 1
+        mags[self.faintObjIdx] = 18.
+        ids[self.faintObjIdx] = "faint"
+        fluxes = np.random.rand(3)
+        self.starStamps = [{"starStamp": starIm, "gaiaGMag": mag, "gaiaId": gaiaId, "annularFlux": flux}
+                           for starIm, mag, gaiaId, flux in zip(starImages, mags, ids, fluxes)]
+        self.innerRadius = 40
+        self.outerRadius = 50
+        self.bss = brightStarStamps.BrightStarStamps(self.starStamps, self.innerRadius, self.outerRadius)
+
+    def tearDown(self):
+        del self.bss
+        del self.starStamps
+        del self.innerRadius
+        del self.outerRadius
+        del self.faintObjIdx
+
+    def testIO(self):
+        """Test the class' write and readFits methods.
+
+        The ``options`` argument to the read method is only used to read
+        sub-BBoxes, which is handled by the Butler. Tests of this are done in
+        afw.
+        """
+        filename = "brightstarStamps.fits"
```

Comment:
You should use the temp file context manager [here](https://docs.python.org/3/library/tempfile.html). This will create you a temporary file in the appropriate place for a given os, and make sure it is cleaned up at the end.